src/Spatial/Euclidean/Quaternion.cs

```diff
@@ -25,53 +24,62 @@ public Quaternion(double real, double imagX, double imagY, double imagZ)
             _y = imagY;
             _z = imagZ;
             _w = real;
-            _norm = ToNorm(real, imagX, imagY, imagZ);
-            _abs = Math.Sqrt(_norm);
-            _arg = Math.Acos(real/_abs);
+            _normSquared = ToNorm(real, imagX, imagY, imagZ);
+            _norm = Math.Sqrt(_normSquared); //TODO : robust _norm calculating
+            _arg = Math.Acos(real / _normSquared);
```

Comment:
Well, how about doing something like this : 
We store those three values in variables after they are calculated by class user.
I am thinking about storing because for example _normSquared is hidden in 

- bool IsUnitQuaternion
- Quaternion Inversed
- Quaternion Ln()

Thus if we create object Quaternion and try to calculate those tree we would calculare it three times!
What I am thinking of calculates it only one : 
Just do backing field for property NormSquared :
```
public double NormSquared{
    get{
        if(normCalculated){ return _normSquared;}
        calculate norm and norm squared
        normCalculated = true
    }
}
```

Or even 
`if(_norm<0) calculate norm and norm squared`
because norm is always >=0